### PR TITLE
gh-103538: Remove unused TK_AQUA code

### PR DESCRIPTION
**Pull request by chrstphrchvz**
* Issue: gh-103538

### CODE REVIEW

**Review by zware — APPROVED** [review]
LGTM, but I realized shortly after merging GH-103535 that it really ought to have a news entry after all, and this one should too.  There's a near-0 chance that anyone is actually using these macros for anything, but we should note that they're gone in any case.

**Comment by chrstphrchvz** [comment]
Is a 2-item ordered list too much for a blurb?

File: Misc/NEWS.d/next/Library/2023-04-14-21-12-32.gh-issue-103538.M4FK_v.rst
Comment (zware) [inline Misc/NEWS.d/next/Library/2023-04-14-21-12-32.gh-issue-103538.M4FK_v.rst]: The bulleted list would probably be fine, but we can condense and add a bit of extra info anyway:

```suggestion
Remove ``_tkinter`` module code guarded by definition of the ``TK_AQUA`` macro
which was only needed for Tk 8.4.7 or earlier and was never actually defined by
any build system or documented for manual use.
```

**Pull request merged**
